### DISCUSSION TITLE
Is there true power on and off command (not toggle)?

### DISCUSSION THREAD

**Discussion by jpjarwis** [supersaiyanmode/PyWebOSTV#64] [Q&A]
I have C1 in PC use. Cause it's tv, it doesn't start/turn off with PC.
I try to simulate monitor behaviour via Windows Task Scheduler. 

I can turn TV off (system.power_off) via Task Scheduler when idle is triggered, but it seems that system.power_off is toggle switch, because Windows triggers idle again and it turns TV on again. Yes, system.power_off command starts my TV when it’s turned off. To solve this, I tried to determine when TV is on or off, like fetching app list (app.list_apps) or read tv info (system.info) or send input (inp.up)… but responds were same when TV was on or off.

Is there true system.power_off() and system.power_on() commands? Or something to try to determine is tv on or off?

There are many problems to solve. Sadly Windows doesn't log sleep, and seems script is too slow to execute when PC powered off (event id 1074). Dirty solution would be to use Raspi to ping PC and turn tv on/off or something like that.

**Comment by supersaiyanmode** [comment]
Did something change in C1? On my CX 2020, system.power_off() works like a Power Off API, not like a toggle. Part of the reason is because once the first power_off is called, the underlying connection is shutdown and further calls don't work. Are you saying the TV still maintains the websocket connection even after powering off (like it's some sort of soft power off)?

Typically, the way it works with my TV (and in that of few others) is:
 - True power off can be called via system.power_off(). TV powers off, underlying connection closed. The library can no longer talk to the TV until it it turned back again. This is why the library doesn't expose a power_on() API.
 - True power on needs to be done via Wake on LAN feature which is outside of the scope of this library but some of the forks have implemented it. This works by sending a magic packet and it uses the TV's mac address.

Can someone else confirm?

**Reply by jpjarwis** [reply]
Now something is changed. Yesterday I tried to find solution for unintentional power-on when PC goes idle, but now it's resolved itself, because it works like you say. No way to communicate with TV when it's turned off. It seems something kept TV "wake" and sockets active even it seemed to be turned off. Last night I tried multiple times (TV was 5 minutes off etc.) and every time TV was reachable and turned on. Shame, I thought was able to turn TV on every time PC is turned on :)

To power-on, maybe the solution is in the future via Bluetooth and Win11 android app support (LG ThinQ).

E: It did it again. It seems TV is doing something in background what prevents full shut down time to time.
